src/analysis/modals/ComputeModal/AzureComputeModal/AzureComputeModal.js

```diff
@@ -83,8 +91,8 @@ export const AzureComputeModalBase = ({
         autopauseThreshold: runtimeDetails ? runtimeDetails.autopauseThreshold || autopauseDisabledValue : defaultAutopauseThreshold,
       });
     });
-    loadCloudEnvironment();
-  });
+    refreshRuntime();
+  }, [currentRuntime, location, onError, workspaceId, setCurrentRuntimeDetails, setComputeConfig]);
```

Comment:
Super nit, but FYI, state setter functions can be left out of effect dependencies because they are stable across renders.
https://legacy.reactjs.org/docs/hooks-reference.html#usestate

```suggestion
  }, [currentRuntime, location, onError, workspaceId]);
```